src/transports/IframeExecutionEnvironmentTransport.ts

```diff
@@ -0,0 +1,84 @@
+import { Duplex } from 'stream';
+import { Transport } from '@open-rpc/client-js/build/transports/Transport';
+import { IJSONRPCData } from '@open-rpc/client-js/build/Request';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import pump from 'pump';
+
+class IframeExecutionEnvironmentTransport extends Transport {
+  private uri: string;
+
+  private stream?: WindowPostMessageStream;
+
+  private commandStream?: Duplex;
+
+  private frame?: Window;
+
+  private postMessageID: string;
+
+  constructor(uri: string) {
+    super();
+    this.uri = uri;
+    this.postMessageID = `post-message-transport-${Math.random()}`;
+  }
+
+  public createWindow(uri: string): Promise<Window> {
+    const iframe = document.createElement('iframe');
+    return new Promise((resolve) => {
+      iframe.addEventListener('load', () => {
+        if (iframe.contentWindow) {
```

Comment:
If this doesn't resolve to `true`, it seems like the Promise would just hang indefinitely